### PR TITLE
Refactoring Configuration

### PR DESCRIPTION
**Pull request by onmyway133**
- This moves things into `Configuration`. There 's `Configuration.default` that applies to every operation. Each operation can overwrite if necessary.

### CODE REVIEW

File: Sources/Shared/ImageManager.swift [hunk 1]
```diff
@@ -20,14 +20,14 @@ public class ImageManager {
   ///   - url: The URL of the image that should be fetched, either from network or cache (if cache is enabled).
   ///   - useCache: Determines if the image should be fetched from cache and stored in cached after fetching.
   ///   - completion: A completion closure that will return the image if it is successfully fetch either from cache or the network.
-  public func fetchImage(from url: URL, useCache: Bool = true, completion: @escaping Completion) {
+  public func fetchImage(from url: URL, configuration: Configuration = Configuration.default, completion: @escaping Completion) {
```
Comment (zenangst) [inline Sources/Shared/ImageManager.swift:23]: @onmyway133 Oh this is pretty neat, this means that you can tailor the configuration per image view right?

File: Sources/iOS/Decompressor.swift [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import UIKit
 
 struct Decompressor {
-  static func decompress(_ data: Data, scale: CGFloat = 1) -> Image {
+  static func decompress(_ data: Data, scale: CGFloat = 1) -> Image? {
     guard let image = Image(data: data) else {
-      return Image()
+      return nil
```
Comment (zenangst) [inline Sources/iOS/Decompressor.swift:6]: This makes sense!

**Review by zenangst — APPROVED** [review]

File: Sources/Shared/ImageManager.swift (hunk 1 above)
Comment (onmyway133) [reply Sources/Shared/ImageManager.swift:23]: @zenangst yeah, we can customise per case, or use default one. We can even customise the cache as well

File: Sources/iOS/Decompressor.swift (hunk 2 above)
Comment (onmyway133) [reply Sources/iOS/Decompressor.swift:6]: ❤️

File: Sources/Shared/ImageManager.swift (hunk 1 above)
Comment (zenangst) [reply Sources/Shared/ImageManager.swift:23]: Pretty darn nifty is if you ask me. Great work! 🚀

Comment (zenangst) [reply Sources/Shared/ImageManager.swift:23]: Having the customization point as the call-site is really great

**Pull request merged**